ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -20,26 +20,66 @@
  * A message info class that contains basic info about a message
  */
 public class MessageInfo {
+
+  public static final short ACCOUNTID_CONTAINERID_DEFAULT_VALUE = -1;
+
   private final StoreKey key;
   private final long size;
-  private final long expirationTimeInMs;
-  private final boolean isDeleted;
-  private final Long crc;
+  private long expirationTimeInMs = Utils.Infinite_Time;
+  private boolean isDeleted = false;
+  private Long crc = null;
+  private short accountId;
+  private short containerId;
+  private long operationTimeMs;
 
-  public MessageInfo(StoreKey key, long size, long expirationTimeInMs) {
-    this(key, size, false, expirationTimeInMs);
-  }
+  public static class Builder {
```

Comment:
Yes, I remember we had a similar discussion during IndexValue. But this time, we do have many callers for MessageInfo : BlobStoreHardDelete, BlobStoreRecovery, ReplicaThread, IndexSegment, PersistentIndex, AmbryRequests and BlobReadOptions. There are 4 constructors used (prior to this patch). With many constructors there are chances of programmatic errors as well. So, I would like to go with builder.